src/readers/FeigReader.java

```diff
@@ -140,6 +140,38 @@ public void onSendProtocol(FedmIscReader arg0, String arg1) {}
 
 	@Override
 	public void onSendProtocol(FedmIscReader arg0, byte[] arg1) {}
+
+	/**
+	 * Read AFI.
+	 *
+	 * @param id
+	 *
+	 * @return
+	 *   AFI
+	 *   -1 == error
+	 */
+	private int readAFI(String id) {
+		fedm.setData(FedmIscReaderID.FEDM_ISC_TMP_B0_REQ_UID, id);
+		fedm.setData(FedmIscReaderID.FEDM_ISC_TMP_B0_CMD, (byte)0x2B);
+		fedm.setData(FedmIscReaderID.FEDM_ISC_TMP_B0_MODE, (byte) 0x00);
+		fedm.setData(FedmIscReaderID.FEDM_ISC_TMP_B0_MODE_ADR, (byte) 0x01);
+
+		try {
+			fedm.sendProtocol((byte) 0xB0);
+
+			int idx = fedm.findTableIndex(0, FedmIscReaderConst.ISO_TABLE, FedmIscReaderConst.DATA_SNR, id);
+			if (idx < 0) {
+				return -1;
+			}
+			byte afi = fedm.getByteTableData(idx, FedmIscReaderConst.ISO_TABLE, FedmIscReaderConst.DATA_AFI);
+
+			return (afi & 0xFF);
+		}
+		catch (Exception e) {
+			System.out.println(e);
```

Comment:
Do we have some possibility to logged exceptions as we would never see this print out in production!